compiler/dyno/lib/uast/chpl-syntax-printer.cpp

```diff
@@ -0,0 +1,1065 @@
+      if (node->withClause()) {
+        printAst(ss_, node->withClause());
+        ss_ << " ";
+      }
+      printBlockWithStyle(ss_, node->blockStyle(), node->stmts());
+    }
+
+    void visit(const Block* node) {
+      printBlockWithStyle(ss_, node->blockStyle(), node->stmts());
+    }
+
+    void visit(const BoolLiteral* node) {
+      ss_ << (node->value() ? "true" : "false");
+    }
+
+    void visit(const BracketLoop* node) {
+      ss_ << "[";
+      if (node->index()) {
+        printAst(ss_, node->index());
+        ss_ << " in ";
+      }
+      printAst(ss_, node->iterand());
+      if (node->withClause()) {
+        ss_<< " ";
+        printAst(ss_, node->withClause());
+      }
+      ss_ << "]";
+      if (node->numStmts() > 0) {
+        ss_ << " ";
+        interpose(ss_, node->stmts(), "");
+      }
+    }
+
+    void visit(const Break* node) {
+      ss_ << "break";
+      if (node->target()) {
+        ss_<< " ";
+        printAst(ss_, node->target());
+      }
+      ss_ << ";";
+    }
+
+    //builder
+    //builderResult
+
+    void visit(const BytesLiteral* node) {
+      ss_ << "b\"" << quoteStringForC(std::string(node->str().c_str())) << '"';
+    }
+
+    void visit(const Catch* node) {
+      ss_ << "catch ";
+      if (node->error()){
+        if (node->hasParensAroundError()) {
+          ss_ << "(";
+          ss_ << node->error()->name().c_str();
+          if (const Expression* te = node->error()->toVariable()->typeExpression()) {
+            ss_ << " : ";
+            printAst(ss_, te);
+          }
+          ss_ << ") ";
+        } else {
+          ss_ << node->error()->name().c_str();
+          if (const Expression* te = node->error()->toVariable()->typeExpression()) {
+            ss_ << " : ";
+            printAst(ss_, te);
+          }
+          ss_ << " ";
+        }
+      }
+      interpose(ss_, node->stmts(), "\n","{", "}");
+    }
+
+    void visit(const Class* node) {
+      ss_ << "class ";
+      ss_ << node->name().c_str() << " ";
+      if (node->parentClass() != nullptr) {
+        ss_ << ": ";
+        printAst(ss_, node->parentClass());
+        ss_ << " ";
+      }
+      interpose(ss_, node->decls(), "\n", "{", "}");
+    }
+
+    void visit(const Cobegin* node) {
+      ss_ << "cobegin ";
+      if (node->withClause()) {
+        printAst(ss_, node->withClause());
+        ss_<< " ";
+      }
+      interpose(ss_, node->taskBodies(), "\n", "{","}");
+    }
+
+    void visit(const Coforall* node) {
+      ss_ << "coforall ";
+
+      if (node->index()) {
+        printAst(ss_, node->index());
+        ss_ << " in ";
+      }
+      printAst(ss_, node->iterand());
+      if (node->withClause()) {
+        ss_<< " ";
+        printAst(ss_, node->withClause());
+      }
+      ss_ << " ";
+      printBlockWithStyle(ss_, node->blockStyle(), node->stmts(), "do ");
+    }
+
+    void visit(const Comment* node){
+      // TODO: Discuss if we want to provide comments back
```

Comment:
It should be relatively painless to write an adaptor over the ASTChildrenIterator which filters out comments. In fact I think we already have one (and you might be able to construct it from the regular iterator with comments).